File: VB/WinSample.Module.Win/ViewHintController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports System.Collections.Generic
Imports System.Diagnostics
Imports System.Text

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Persistent.Base
Imports DevExpress.ExpressApp.Templates
Imports System.Windows.Forms
Imports DevExpress.Utils.Frames

Namespace WinSample.Module.Win
	Partial Public Class ViewHintController
		Inherits ViewController
		Public Sub New()
			InitializeComponent()
			RegisterActions(components)
			TargetViewType = ViewType.ListView
		End Sub
		Protected Overrides Sub OnActivated()
			MyBase.OnActivated()
			AddHandler Frame.GetController(Of WindowHintController)().HintPanelReady, AddressOf ViewHintController_HintPanelReady
		End Sub
		Protected Overrides Sub OnDeactivated()
			MyBase.OnDeactivated()
			RemoveHandler Frame.GetController(Of WindowHintController)().HintPanelReady, AddressOf ViewHintController_HintPanelReady
		End Sub
		Private Sub ViewHintController_HintPanelReady(ByVal sender As Object, ByVal e As HintPanelReadyEventArgs)
			e.HintPanel.Text = View.Id
			e.HintPanel.Visible = True
		End Sub
	End Class
End Namespace

File: CS/WinSample.Module.Win/ViewHintController.cs
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.Base;
using DevExpress.ExpressApp.Templates;
using System.Windows.Forms;
using DevExpress.Utils.Frames;

namespace WinSample.Module.Win {
    public partial class ViewHintController : ViewController {
        public ViewHintController() {
            InitializeComponent();
            RegisterActions(components);
            TargetViewType = ViewType.ListView;
        }
        protected override void OnActivated() {
            base.OnActivated();
            Frame.GetController<WindowHintController>().HintPanelReady += new EventHandler<HintPanelReadyEventArgs>(ViewHintController_HintPanelReady);
        }
        protected override void OnDeactivated() {
            base.OnDeactivated();
            Frame.GetController<WindowHintController>().HintPanelReady -= new EventHandler<HintPanelReadyEventArgs>(ViewHintController_HintPanelReady);
        }
        void ViewHintController_HintPanelReady(object sender, HintPanelReadyEventArgs e) {
            e.HintPanel.Text = View.Id;
            e.HintPanel.Visible = true;
        }
    }
}
